app_grim_reefer/src/main.c

```diff
@@ -27,54 +47,168 @@ const struct gpio_dt_spec led2 = GPIO_DT_SPEC_GET(LED2_NODE, gpios);
         LOG_ERR("Unable to configure LED 1 output pin\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
+
     if (!gpio_is_ready_dt(&led2)) {
         LOG_ERR("LED 2 is not ready\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
     if (gpio_pin_configure_dt(&led2, GPIO_OUTPUT_ACTIVE) < 0) {
         LOG_ERR("Unable to configure LED 2 output pin\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
     // Init Enable pins
     if (!gpio_is_ready_dt(&ldo_enable)) {
         LOG_ERR("ldo enable pin is not ready\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
     if (gpio_pin_configure_dt(&ldo_enable, GPIO_OUTPUT_ACTIVE) < 0) {
         LOG_ERR("Unable to configure ldo enable output pin\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
+
     if (!gpio_is_ready_dt(&cam_enable)) {
         LOG_ERR("camera enable pin is not ready\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
     if (gpio_pin_configure_dt(&cam_enable, GPIO_OUTPUT_ACTIVE) < 0) {
         LOG_ERR("Unable to configure camera enable output pin\n");
-        return -1;
+        return INIT_GPIO_FAIL;
+    }
+
+    if (!gpio_is_ready_dt(&buzzer)) {
+        LOG_ERR("buzzer pin is not ready\n");
+        return INIT_GPIO_FAIL;
+    }
+    if (gpio_pin_configure_dt(&buzzer, GPIO_OUTPUT_ACTIVE) < 0) {
+        LOG_ERR("Unable to configure buzzer output pin\n");
+        return INIT_GPIO_FAIL;
+    }
+
+    if (!device_is_ready(debug_serial_dev)) {
+        LOG_ERR("Debug serial not ready\n");
+        return INIT_GPIO_FAIL;
     }
+
+    gpio_pin_set_dt(&led1, 0);
+    gpio_pin_set_dt(&led2, 0);
+    gpio_pin_set_dt(&ldo_enable, 1);
+    gpio_pin_set_dt(&cam_enable, 0);
+    gpio_pin_set_dt(&buzzer, 0);
+
+    return INIT_OK;
+}
+
+static int sensor_init(void) {
+    const bool flash_found = device_is_ready(flash_dev);
+    if (!flash_found) {
+        return INIT_NOFLASH;
+    }
+
+    const bool lsm6dsl_found = device_is_ready(lsm6dsl_dev);
+    const bool bme280_found = device_is_ready(bme280_dev);
+    if (!lsm6dsl_found) {
+        LOG_ERR("Error setting up LSM6DSL");
+        return INIT_MISSING_SENSOR;
+    }
+    if (!bme280_found) {
+        LOG_ERR("Error setting up BME280");
+        return INIT_MISSING_SENSOR;
+    }
+    const bool ina_bat_found = device_is_ready(ina_bat_dev);
+    const bool ina_ldo_found = device_is_ready(ina_ldo_dev);
+    const bool ina_grim_found = device_is_ready(ina_grim_dev);
+
+    if (!ina_bat_found || !ina_ldo_found || !ina_grim_found) {
+        LOG_ERR("Error setting up INA260 devices");
+        return INIT_MISSING_SENSOR;
+    }
+
+    // ADC
+    if (!adc_is_ready_dt(&adc_chan0)) {
+        LOG_ERR("ADC controller device %s not ready\n", adc_chan0.dev->name);
+        return INIT_MISSING_SENSOR;
+    }
+    //
+    if (adc_channel_setup_dt(&adc_chan0) < 0) {
+        LOG_ERR("Could not setup ADC channel\n");
+        return INIT_MISSING_SENSOR;
```

Comment:
Think we can avoid re-inventing the wheel and use Zephyr's errno constants. For example these returns would be an -ENODEV